include/RED4ext/StringView.hpp

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <RED4ext/Common.hpp>
+
+#include <string_view>
+
+namespace RED4ext
+{
+struct CString;
+
+#pragma pack(push, 4)
+struct StringView
+{
+    constexpr StringView() noexcept;
+    constexpr StringView(const char* aStr) noexcept;
+    constexpr StringView(std::string_view aView) noexcept;
+    StringView(const RED4ext::CString& aStr) noexcept;
+
+    constexpr bool IsValid() const noexcept;
+    constexpr operator bool() const noexcept;
+
+    constexpr bool operator==(const StringView& aRhs) const noexcept;
```

Comment:
as you seem to offer only equality, i would probably consider adding compare function returning an integer so one can potentially sort the views in some way

also could potentially allow for use of spaceship operator to define all of the comparisons, but that is extra i suppose...